File: rosettacode_org/C/Copy_a_string/Copy_a_string.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    char src[] = "Hello";
    char dst[80];

    /* Use strlcpy() from <string.h>. */
    if (strncpy(dst, src, sizeof dst) >= sizeof dst) {
        fputs("The buffer is too smaill!\n", stderr);
        exit(1);
    }

    memset(src, '-', 5);
    printf("src: %s\n", src); /* src: ----- */
    printf("dst: %s\n", dst); /* dst: Hello */

    return 0;
}

File: rosettacode_org/C/Copy_a_string/Copy_a_string.py
src = "hello"
a = src
b = src[:]
print(a)
print(b)
import copy
c = copy.copy(src)
d = copy.deepcopy(src)
t = src is a is b is c is d
print(t)

a = "hello"
b = ''.join(a)
print(a == b)
print(b is a)
